agents1/OfficialAgent.py

```diff
@@ -402,36 +420,44 @@ def decide_on_actions(self, state):
+
+                        # Determine the next area to explore if the human tells the agent not to remove the obstacle
                         if self.received_messages_content and self.received_messages_content[
                             -1] == 'Continue' and not self._remove:
                             self._answered = True
                             self._waiting = False
+
                             # Add area to the to do list
                             self._tosearch.append(self._door['room_name'])
                             self._phase = Phase.FIND_NEXT_GOAL
+
                         # Wait for the human to help removing the obstacle and remove the obstacle together
-                        if self.received_messages_content and self.received_messages_content[
-                            -1] == 'Remove' or self._remove:
+                        if self.received_messages_content and self.received_messages_content[-1] == 'Remove' or self._remove:
                             if not self._remove:
                                 self._answered = True
-                            # Tell the human to come over and be idle untill human arrives
+
+                            # Tell the human to come over and be idle until human arrives
                             if not state[{'is_human_agent': True}]:
-                                self._sendMessage('Please come to ' + str(self._door['room_name']) + ' to remove rock.',
-                                                  'RescueBot')
+                                self._sendMessage(f'Please come to {self._door["room_name"]} to remove rock.', 'RescueBot')
                                 return None, {}
+                            
                             # Tell the human to remove the obstacle when he/she arrives
                             if state[{'is_human_agent': True}]:
-                                self._sendMessage('Lets remove rock blocking ' + str(self._door['room_name']) + '!',
-                                                  'RescueBot')
+                                self._sendMessage(f'Lets remove rock blocking {self._door["room_name"]}!', 'RescueBot')
                                 return None, {}
-                        # Remain idle untill the human communicates what to do with the identified obstacle
+
+                        # Remain idle until the human communicates what to do with the identified obstacle
                         else:
                             return None, {}
 
-                    if 'class_inheritance' in info and 'ObstacleObject' in info['class_inheritance'] and 'tree' in info[
-                        'obj_id']:
+                    if 'class_inheritance' in info and 'ObstacleObject' in info['class_inheritance'] and 'tree' in info['obj_id']:
                         objects.append(info)
+
+                        # TODO: Do we need to remove trust when there's a tree? (Robot can do it on its own)
```

Comment:
Don't think we need to remove trust when there is a tree,
the human is asking the robot to remove an obstacle and the tree can only be removed by rescuebot itself
Since the human does not need to help he doesn't have to stay in my opinion